### ISSUE TITLE
User Interface to create and manage scheduled toot's

### ISSUE THREAD

**Issue by schiessle** [tootsuite/mastodon#10088]
The last release implemented the back-end and API to enable scheduled toot's https://github.com/tootsuite/mastodon/pull/9706. That's great, thanks a lot @Gargron. The first Apps like Fedilab on Android already use the new API. But I think in order to be really useful, the user needs to be able to create scheduled toot's in the web interface and manage already scheduled toot's.

**Comment by resynth1943** [comment]
Has anyone claimed this?

**Comment by leahoswald** [comment]
Still missing this feature.

**Comment by Goldmaster** [comment]
Really is needed. Many brands and celebrities are joining and the ability to schedule toots would be highly useful in the web and mobile app. For now I'm hoping https://www.scheduler.mastodon.tools/ will work until scheduling gets added.

**Comment by eloquence** [comment]
Yes, this is a key feature esp. for organizations as more of civil society joins the fediverse. The third party tool appears to work fine for now (and you can run it locally), but this really is a core feature.

The UI could also make sure that users don't hit the surprising "if it's less than 5 minutes, it will be posted immediately" behavior (#9884).

**Comment by FediVideos** [comment]
Is there some reason there isn't an official interface for this? Have had queries asking how to schedule toots, feels a bit odd telling them there's an official scheduler but no interface.

**Comment by goodevilgenius** [comment]
> For now I'm hoping [scheduler.mastodon.tools](https://www.scheduler.mastodon.tools/) will work until scheduling gets added.

There's also https://plan.fedilab.app/

**Comment by SilverBut** [comment]
well, i'm working on my own branch. maybe a pull request tomorrow.

**Comment by SilverBut** [comment]
I have finished the part of sending scheduled toots. But managing them is not that easy. I'm not a senior developer for mastodon so I think I should seek some advice from the community.

I'm reusing the ordinary timeline and components for ordinary toots. However, the [scheduled toots](https://docs.joinmastodon.org/entities/ScheduledStatus/) actually lacks some fields in ordinary toots, [for example](https://docs.joinmastodon.org/entities/Status/#content), `content`, which is the rendered text. There are also some other differences including:

* Scheduled toots are placed in `params` while ordinary toots are placed at root
* Scheduled toots have no account information (which seems reasonable because one can only see their own schedule)
* Text for screen reader would be different from ordinary statuses
* etc.

Well, I have no exact idea about if I should write some new components. Currently I'm changing on existing components, but sometime this makes old code looks weird. Is there any guideline on this?

Also I think several fields including `content` are necessary to display contents on the webpage, while missing from scheduled toot's API.  After a quick glance at #9706 I'm still not sure the reason why the API is designed like this. Should there be a proposal on changing API for scheduled toots to make the `[ScheduledStatus](https://docs.joinmastodon.org/entities/ScheduledStatus/)` having more similarity with [Status](https://docs.joinmastodon.org/entities/Status/), so both official & 3rdparty clients can reuse more code?

**Comment by Goldmaster** [comment]
Where are we upto with this? Wish this could be done inside the official app and on the Web interface. Such a feature would blow Facebook threads out of the water.

**Comment by rkingett** [comment]
Putting in my own two cents and saying I also would love the ability to select days or even months into the future.

**Comment by steodor** [comment]
+1 for this to be main functionality rather than via external apps or services. Thanks!

**Comment by SlippingGitty** [comment]
+1 one to say I would appreciate this feature, too

**Comment by EmilJacobs** [comment]
I've been using publer.io for exactly this feature, which is a paid service. That Mastodon has the capability but just lacks the UI to actually use it just blows my mind.

**Comment by steodor** [comment]
@EmilJacobs thanks, actually the 10 scheduled posts in the free tier would be enough for me. But yeah, there's one hint as to why Mastodon have apparently stopped development mid-way on this. I don't get it, social media tools have way more features and crucially account centralization, which means a feature such as this in Mastodon wouldn't compete, not by far.

**Comment by hex-m** [comment]
For people that look for a (proprietary and paid) workaround there is also [SocialHub](https://socialhub.io/en/). (I'm not affiliated and I would prefer this feature to come with Mastodon)

**Comment by jeremiahlee** [comment]
For reference, this is the UI on Twitter:

1. Tap schedule icon (calendar and clock) from compose tweet UI.
2. Schedule modal appears. Select the date and time. Tap confirm. ![Image](https://github.com/user-attachments/assets/656e8918-53c2-4822-88f8-2ddda8ecf755)
3. Schedule modal closes. Post button becomes Schedule button on the compose tweet UI. ![Image](https://github.com/user-attachments/assets/3ca63619-3b9a-4a59-a44f-d882934eaeb3)

View and edit scheduled tweets by:
1. Tap schedule icon (calendar and clock) from compose tweet UI.
2. Schedule modal appears. Tap Scheduled posts link.
3. Drafts modal appears and shows scheduled posts. Tap scheduled post to edit. ![Image](https://github.com/user-attachments/assets/76dcde87-db46-45a6-b486-76bb73f88866)
4. Al modals close and scheduled tweet is show in the compose tweet UI.  Post button becomes Schedule button.

**Comment by nekohayo** [comment]
For those running Linux: [Tuba](https://flathub.org/apps/dev.geopjr.Tuba) now supports this.  
Its UI for creating and managing/editing/rescheduling scheduled posts is pretty much flawless, it does everything I've needed it to. Maybe it can serve as inspiration for Mastodon's web and mobile interfaces.

**Comment by Dret42** [comment]
Please, implement this feature as soon as possible!!! 
Thanks!

**Comment by tyradux** [comment]
Oh yes, please add this feature to the web interface!

**Comment by renchap** [comment]
We want to do this, but it will require significant changes to the Mastodon web interface this is not something that we feel like an external contributor can do.

The main challenge here is that the composer does not have a lot of real estate in the interface and adding more features to it is complicated.

But we intend to tackle this for our next version (tentatively planned for end of summer) where the composer will move somewhere else, which will allow space to add such features to the web interface. We have not started working on this yet (we need to ship Mastodon 4.6 first) and plans can change, but this is definitely near the top of the things we want to do for the next version.

**Comment by rauschma** [comment]
@renchap Excellent news! One option is to simply add an item “Scheduled posts” to the main menu (e.g. next to “Bookmarks”).

* Upside: A clear location where you can look up what’s currently scheduled.
* Downside: Not as easy to discover as a button in the post composer.

**Comment by mitchellecm7** [comment]
/claim #10088
Working on the web UI for creating and managing scheduled toots via PR #38973.

**Comment by Mieruko** [comment]
So, is that bounty still valid?

**Comment by nschonni** [comment]
@dhuschde the bounty you created is just attracting spam PRs. If you want to keep it as an incentive to award to the Mastodon devs if this lands, you might want to change the status so that it doesn't seem like it's actually open for random developers.

**Comment by meowIAmAUser** [comment]
I am sorry but i can't see any way to do this. Otherwise i would have done it way earlier. If I knew what this would lead to i would have never created the Bounty.

**Comment by EmilJacobs** [comment]
> I am sorry but i can't see any way to do this. Otherwise i would have done it way earlier. If I knew what this would lead to i would have never created the Bounty.

Honestly, if a baked in feature of Github is so easily exploited by spam PRs, it's not your fault here. It's a bug. Plain and simple.

Anyway, I'm willing to retract my bounty pledge if it creates these kinds of issues. If that helps.

**Comment by omarsoufiane** [comment]
Hello, I'm from bountyhub, recently we have a lot of bots that scan github to make ai generated pr, and we don't currently have a solution for it.
Since this bounty is no longer available and would be handled by the mastodon team, I unlisted it.

those who made contributions would be refunded.

**Comment by mitre88** [comment]
Hi there — I saw this bounty and I'd like to take a shot at it. I'll fork the repository, implement a focused fix, and open a PR referencing this issue with a clear description of the change and how it was verified. I'm comfortable iterating on review feedback. Thanks!

**Comment by renchap** [comment]
@mitre88 Please do not. As indicated above, we will not accept contributions for this feature as we are already working on it alongside a large composer rework.
